### ISSUE TITLE
prefer-for-of false positive

### ISSUE THREAD

**Issue by Paril** [mysticatea/eslint-plugin#25]
The following pattern causes prefer-for-of to trip even though it's impossible to turn the for into a for-of.

```
const a = { b: 5 };
const b = { c: 5 };

for (const key in a)
	if (!(key in b))
		b[key] = a[key];
```

I'm sure there might be a better way of performing this pattern without the for (maybe Object.assign?), but just letting you know.

**Comment by Paril** [comment]
Oh, I see, it wants me to use Object.keys rather than "in". My bad. The error msg should probably reflect that, though.
